python/src/aac/spec/spec.yaml

```diff
@@ -357,6 +357,8 @@ data:
         instances of the modeled entity to support horizontal scaling.
   required:
     - name
+  validation:
+    - name: Subcomponents are models
```

Comment:
I wonder if we could provide validation args and kwargs simultaneously. I didn't like how the just kwargs looked, it's pretty clunky, but if we also supported kwargs (Maybe mutually exclusive with just *args?), then we could make this validation more general in the future.

For example:
```yaml
  validation:
    - name: Field is type
       arguments:
           - name: field names
              args: 
                  - components
           - name: allowed types
              args:
                  - model     
```

Reply:
It seems to be a pretty common pattern to use `(..., *args, **kwargs)` in parameter lists that take variable numbers of arguments, so I'd be in favor of allowing both, when they're needed. To my way of thinking, that would be the way to keep things as general as possible.